services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusControllerITest.kt

```diff
@@ -1132,13 +1136,59 @@ class HakemusControllerITest(@Autowired override val mockMvc: MockMvc) : Control
             every {
                 authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
             } returns true
-            every { hakemusService.sendHakemus(id, USERNAME) } returns hakemus
+            every { hakemusService.sendHakemus(id, null, USERNAME) } returns hakemus
 
             post(url).andExpect(status().isOk).andVerifyRegistryKeys()
 
             verifySequence {
                 authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
-                hakemusService.sendHakemus(id, USERNAME)
+                hakemusService.sendHakemus(id, null, USERNAME)
+            }
+        }
+
+        @Test
+        fun `returns 400 when request body is missing fields`() {
+            val paperDecisionReceiver = PaperDecisionReceiverFactory.default
+            val request = HakemusSendRequest(paperDecisionReceiver = paperDecisionReceiver)
+            val json: ObjectNode = OBJECT_MAPPER.valueToTree(request)
+            val child = json.get("paperDecisionReceiver") as ObjectNode
+            child.remove("name")
+
+            post(url, json).andExpect(status().isBadRequest())
+        }
+
+        @Test
+        fun `returns paper decision receiver when request body is present`() {
+            val paperDecisionReceiver = PaperDecisionReceiverFactory.default
+            val request = HakemusSendRequest(paperDecisionReceiver = paperDecisionReceiver)
+            every {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+            } returns true
+            every { hakemusService.sendHakemus(id, any(), USERNAME) } returns
```

Comment:
Should here be used `paperDecisionReceiver` as 2nd parameter instead of `any()`?

Reply:
I changed this to use `paperDecisionReceiver` to keep up the pattern in other tests.